Update get started with human task orchestration to include Play

mesellings opened the pull request:
## Description

Update the https://docs.camunda.io/docs/next/guides/orchestrate-human-tasks/ topic to include Play, milestone creation, and review as steps before running it on the cluster.

Relates to issue https://github.com/camunda/camunda-docs/issues/4183.

<!-- Provide an overview of what to expect in the PR. -->
<!-- Link to an associated epic when applicable. -->
<!-- Add an assignee and use `component:` or version labels for good hygiene! -->

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [X] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

<!-- Camunda maintains 18 months of versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [X] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

<!-- All changes require either an Engineering review or technical writer review. **Many require both!** -->

- [X] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [X] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). -->

conceptualshark reviewed (changes requested):
A few small notes - I also unfortunately!! overhauled this page recently to update the self-managed instructions, so it might need some work to align it after the merge conflicts are resolved. 

Play is not available in the Desktop Modeler-based SM version of the guides, so you shouldn't need to add it to those new sections.

mesellings commented:
@conceptualshark Don't worry, I'm going to rebase this and start again on Monday - I saw your changes, so I'll need to adapt the new stuff for this - don't spend any more time on it 😄

akeller commented:
Removing `deploy` since it's currently broken. 😢

mesellings commented:
This needs a review by Eric, so should not be merged - was waiting for the deploy workflow to be fixed - we can hopefully push this tomorrow

mesellings commented:
@HanselIdes The preview environment is (finally) fixed and up for you to review: 

* https://preview.docs.camunda.cloud/pr-4401/docs/next/guides/orchestrate-human-tasks/

Depending on when you are able to review, we can try to get this into the release tomorrow, or push it afterwards?

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -23,21 +25,26 @@ import RunProcessSM from './img/run-process-sm.png';
 import Install from './react-components/install-c8run.md'
 
-Camunda 8 allows you to orchestrate processes with human tasks of any complexity. Utilizing user tasks, you can create and assign tasks to users. Then, users can perform their work and enter the necessary data to drive the business process.
+You can use Camunda 8 to orchestrate processes with human tasks of any complexity. User tasks allow you to create and assign tasks to users, who can then perform their work and enter the required data to drive the business process.
```
```suggestion
You can use Camunda 8 to orchestrate processes with human tasks of any complexity. Use the user task BPMN element to create and assign tasks to users, who can then perform their work and enter the required data to drive the business process.
```
For the getting started with HTO guide, I'd also be interested in seeing @volodymyr-melnykc 's perspective as he's the HTO PM

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -100,13 +139,14 @@ To run this guide, make sure to be in **Implement** mode to specify the technica
 
 1. A **start event** is automatically added to the canvas. Click it to display configuration and append options.
```
The start event should be automatically selected- you can check this behavior on SaaS

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -100,13 +139,14 @@ To run this guide, make sure to be in **Implement** mode to specify the technica
 
 1. A **start event** is automatically added to the canvas. Click it to display configuration and append options.
 2. Click the rectangular **Append Task** icon to append a task.
```
<img width="806" alt="image" src="https://github.com/user-attachments/assets/f0d29da9-95ba-4212-9c21-dfc79c921919">
These instructions should be updated to account for the new context pad. The user has to hover over the blue circle to the right, then click the rectangular append task icon. This applies for all instructions- they should be double-checked

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -120,48 +160,47 @@ Variables are part of a process instance and represent the data of the instance.
-2. Click the blue **link icon** in the lower right corner. A menu expands that allows you to create a new form.
-   <img src={ModelerFormMenuImg} style={{width: 400}} alt="Annotation to open the form menu" />
-3. Click **Create new form**. A form will be created and opened in the form editor. The form is automatically named.
+1. Click the blue **link icon** in the lower right corner. A menu expands that allows you to create a new form.
```
<img width="432" alt="image" src="https://github.com/user-attachments/assets/dae13207-660c-471f-b82f-2f5d40ad61cc">
I saw this instead of the empty state

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -174,7 +213,7 @@ If the properties panel for your form doesn't open automatically, navigate to **
 
 ## Step 3: Link the form to your process
```
If you create a form according to these instructions, then this step is automatically completed.

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -58,21 +65,49 @@ This guide introduces you to the basics of human task orchestration. You will cr
+First, you must create the BPMN diagram file and project.
+
+1. Open Web Modeler, and click **New project**. Every file in Web Modeler requires a project.
+2. Select **Create new > BPMN diagram**.
```
For automation, users should always use process applications. They have the most features, including a smoother Play deployment experience

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -206,29 +249,54 @@ If the properties panel for your task doesn't open automatically, navigate to **
 
-### Deploy and test run
+1. Click the **Play** tab to enter Play mode.
+1. Once the Play environment is ready, click **Start a process instance** to start testing your process.
```
With 8.6, the user will need a cluster rather than just waiting for the Play environment.
![image](https://github.com/user-attachments/assets/c9c659e3-9afb-41cf-b7d2-d2deaa640df8)
![image](https://github.com/user-attachments/assets/840778c7-7365-4a94-8c67-5e8575e44746)
![image](https://github.com/user-attachments/assets/145e1d21-4ffa-4cc2-933c-5ea63ac73083)
![image](https://github.com/user-attachments/assets/e6b10b33-6ea3-4834-83c6-d42433029b6c)

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -206,29 +249,54 @@ If the properties panel for your task doesn't open automatically, navigate to **
+As well as using Play mode to quickly validate and run your process in development, you can also:
+
+- Deploy the process to a [cluster](/components/concepts/clusters.md) in other environments such as testing, staging, and production. After you deploy your process, it can be run on the cluster.
+- Run and complete the user task in [Tasklist](/components/tasklist/introduction-to-tasklist.md). Applications such as Tasklist can be used by humans to complete tasks. As well as using Play mode and Tasklist to run a process, you can call the process via the API or an inbound trigger. Read more about [run options](/components/modeler/web-modeler/run-or-publish-your-process.md).
```
You can follow a link to Operate...
![image](https://github.com/user-attachments/assets/42c511d3-7748-419a-859c-ac579a640525)
and then a link to Tasklist if you have an open user task...
![image](https://github.com/user-attachments/assets/748418bc-906f-4d0b-82e2-5f8cf6ace7fb)

Or you can open Tasklist and it should show the open task directly

HanselIdes reviewed (changes requested):
This is good progress, but fundamental updates to the user journey and differences between Desktop Modeler+C8 Run, Web Modeler SM, and Web Modeler SaaS make it difficult to follow the changes. It's probably best to group the tabs as
Desktop Modeler | Web Modeler
especially because we're uniting the UX between Web Modeler SaaS and SM

This complicates the flow and also the table of contents, where there are two step 4s
![image](https://github.com/user-attachments/assets/5c5efd52-b868-4aec-891e-f94456309ff5)

mesellings commented:
> This is good progress, but fundamental updates to the user journey and differences between Desktop Modeler+C8 Run, Web Modeler SM, and Web Modeler SaaS make it difficult to follow the changes. It's probably best to group the tabs as Desktop Modeler | Web Modeler especially because we're uniting the UX between Web Modeler SaaS and SM
> 
> This complicates the flow and also the table of contents, where there are two step 4s ![image](https://private-user-images.githubusercontent.com/17521551/374460472-5c5efd52-b868-4aec-891e-f94456309ff5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzODUxMTksIm5iZiI6MTcyODM4NDgxOSwicGF0aCI6Ii8xNzUyMTU1MS8zNzQ0NjA0NzItNWM1ZWZkNTItYjg2OC00YWVjLTg5MWUtZjk0NDU2MzA5ZmY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA4VDEwNTMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmZDhiNTZmNzg3M2UwYzg5YTU3NTA0NjMyZGNkN2M1MTI0ZmMxOTQ1MmM3Y2ExNDVmYzlkZTY0MjRhYmE1NTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.L2mAIhoQbqpPgkcsXZqgMktsgttqRxYFpZ897i_V_S4)

Thanks @HanselIdes I'll take another look at this - I really appreciate the comprehensive feedback 👍

mesellings commented:
@HanselIdes I've gone through the SaaS section and incorporated all your feedback - please check you are happy with the new structure, and I will move onto the SM section.

https://preview.docs.camunda.cloud/pr-4401/docs/next/guides/orchestrate-human-tasks/

- I've condensed it into 3 stages: Step 1: Create and design a process, Step 2: Design and link a form, and Step 3: Run and validate your process.
- I think it might be good to keep the SaaS & Web Modeler / SM Camunda 8 Run & Desktop Modeler split as it covers multiple bases for users - what do you think?

p.s. Ignore the right-hand menu for now, as it is still affected by the unchanged SM content.

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -158,88 +176,60 @@ Variables are part of a process instance and represent the data of the instance.
+:::
 
+1. Select the user task you created in **[Step 1](#step-1-create-a-new-process)**.
+1. Click the blue **Link Form** icon in the context pad, and click **Create new form**.
```
This is the first time the "context pad" is mentioned, but there's nothing showing what it looks like or where to find it. The same thing applies to the palette. @lmbateman are these intuitive enough concepts that we can assume the user knows what we're talking about?

mesellings replied:
I've added a link to the context pad page here 👍

HanselIdes reviewed (changes requested):
The one thing that must be addressed is the inaccuracy between SaaS and SM. In this guide as it's written, SM represents local development (Desktop Modeler + C8 Run), not self-managed.

mesellings commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -28,19 +28,26 @@ import NavigationHistoryImg from './img/modeler-navigation-history.png';
 
 This guide introduces you to the basics of human task orchestration. You will create a simple process to decide on what to eat for dinner, and drive the process flow according to that decision.
 
+- Select **Self-Managed** to see an example using Camunda 8 Run and Desktop Modeler.
```
Thanks, I understand now, appreciate the clarification - I have changed to Desktop and Web (to match the download language), and updated the intro:

![image](https://github.com/user-attachments/assets/8930d4d8-6699-4e8c-b2c7-a16bbb3770ba)

mesellings commented:
@HanselIdes I've made those changes now, and other than the Append element comments, this is probably good to go for a final review? 

In summary, I changed language to Desktop/Web as suggested, added a context pad link, and removed some heading elements from the Desktop validate section to avoid them appearing in the right-hand TOC (until we can fix this so it takes tabbed content into context).

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -45,17 +45,17 @@ You can use Camunda 8 to orchestrate processes with human tasks of any complexit
-- Select **Self-Managed** to see an example using Camunda 8 Run and Desktop Modeler.
-- Select **SaaS** to see an example using Camunda 8 SaaS and Web Modeler.
+- Select **Desktop** to see an example using Camunda 8 Run and Desktop Modeler in a local development environment.
+- Select **Web** to see an example using Camunda 8 SaaS and Web Modeler.
```
Web Modeler Self-Managed is excluded, but that's appropriate for this stage of the journey

HanselIdes reviewed (approved):
LGTM!

conceptualshark commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -58,21 +72,56 @@ This guide introduces you to the basics of human task orchestration. You will cr
+## Step 1: Create and design a process
 
-### Create a new file
+Start by creating and designing a process to demonstrate how to route the process flow based on a user decision. In this example, you will create a process to decide what to eat for dinner.
```
```suggestion
To learn how to route a process flow based on a user decision, you will create and design a process to decide what to eat for dinner.
```

conceptualshark commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -98,18 +151,22 @@ To run this guide, make sure to be in **Implement** mode to specify the technica
-5. Select the user task and click on the diamond-shaped icon to append an exclusive gateway. The gateway allows to route the process flow differently, depending on conditions.
-6. Select the gateway and append a task by clicking the task icon. Repeat it to create a second process flow. Name the tasks based on what the user decides to eat: in this case, we've named ours `Prepare chicken` and `Prepare salad`.
+1. A **start event** is automatically added to the canvas and selected.
+2. Append a task. For example, hover over the blue plus icon and select the **Append Task** icon from the [context pad](/components/modeler/web-modeler/context-pad.md).
```
Context pad is more useful for the SaaS/web modeler side of things and not the same experience in desktop. If this section is swapped to `Append element` instead (and the context pad reference removed), it brings it back in line with the Desktop experience and would remove the need for another set of split/tabbed content.

conceptualshark commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -118,116 +175,112 @@ Variables are part of a process instance and represent the data of the instance.
-:::note
-If the properties panel for your form doesn't open automatically, navigate to **Window -> Toggle Properties Panel** to open it manually.
-:::
+1. Open the process you created in **[Step 1](#step-1-create-a-new-process)** by clicking on the process file's name in the top bar.
```
```suggestion
1. Open the process you created in **[Step 1](#step-1-create-and-design-a-process)** by clicking on the process file's name in the top bar.
```

conceptualshark commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -118,116 +175,112 @@ Variables are part of a process instance and represent the data of the instance.
-:::
+1. Open the process you created in **[Step 1](#step-1-create-a-new-process)** by clicking on the process file's name in the top bar.
+2. Select the user task, and open the **Form** menu in the properties panel.
+3. In the Form menu, enter the **Form ID** for the form you created in **[Step 2](#step-2-design-a-form)**.
```
```suggestion
3. In the Form menu, enter the **Form ID** for the form you created in **[Step 2](#step-2-design-and-link-a-form)**.
```

conceptualshark commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -240,64 +293,30 @@ Ensure your installation of [Camunda 8 Run](/self-managed/setup/deploy/local/c8r
-</Tabs>
-
 2. On the left, you will notice a list of **tasks**. There should be one open task `Decide what's for dinner`. Click this task to open it in the detail view.
 3. In the detail view, the form you created in **[Step 2](#step-2-design-a-form)** appears. It is read only since this task is currently unassigned. You have to claim the task to work on it. Next to **Assignee**, click **Assign to me** to claim the task.
```
```suggestion
3. In the detail view, the form you created in **[Step 2](#step-2-design-and-link-a-form)** appears. It is read only since this task is currently unassigned. You have to claim the task to work on it. Next to **Assignee**, click **Assign to me** to claim the task.
```

conceptualshark reviewed (changes requested):
My lingering question here is why the addition of Play changes the Web Modeler section of this guide into "validate with Play" as the end result, and not completing and checking on the task in Tasklist and Operate. 

I don't think this guide should solely walk a user through building a model and using Play for initial validation. Presenting options for understanding how that process moves through/works with Operate and Tasklist is an equally important end result.

HanselIdes commented:
> My lingering question here is why the addition of Play changes the Web Modeler section of this guide into "validate with Play" as the end result, and not completing and checking on the task in Tasklist and Operate.
> 
> I don't think this guide should solely walk a user through building a model and using Play for initial validation. Presenting options for understanding how that process moves through/works with Operate and Tasklist is an equally important end result.

@conceptualshark The goal mentioned in the guide says:
"This guide introduces you to the basics of human task orchestration. You will create a simple process to decide on what to eat for dinner, and drive the process flow according to that decision."

The best way to "drive the process flow" is through Play. It requires no navigation, and the additional features that Tasklist has (assignee, priority, process visualization) are unnecessary and perhaps distracting. I think it makes sense to _mention_ that you can also go through Operate and Tasklist, but it is not necessary to achieve the stated goal.

If you believe the "get started" guides should feature Operate and Tasklist, then we can talk about changing their goal. Within the PM team, the focus has been around getting to 'hello world'. When I learned Python scripting (which is my background- I'm not a pro dev here), "hello world" happens in the console within a second of writing or editing the code with as little context change as possible.

mesellings commented:
@HanselIdes @conceptualshark Thanks for the discussion, I think we should see whether @volodymyr-melnykc has a preferred perspective on this as well for HTA.

@volodymyr-melnykc - if the main goal of the Getting started with HTA doc is to provide a quick introduction into integrating HTA into your processes, should the path for this (SaaS) be:
* Use Play to show how you can add and simulate using user task forms, and simply refer to using Tasklist & Operate.
OR
* As per the Desktop Modeler & Camunda 8 Run path, show how to use Tasklist and Operate (and not Play).

From my perspective, this hinges around what type of guide this is meant to be - is it an exhaustive 'this is how to do it', or more of a worked example 'here is an overview of how you can do it' that aims to introduce the key concepts without making a new user open multiple applications to begin with (i.e. keeping it simple)? It sounds like the preferred path is to get to "Hello World" as quickly/easily as possible, in which case Play makes sense?

conceptualshark commented:
> As per the Desktop Modeler & Camunda 8 Run path, show how to use Tasklist and Operate (and not Play).

I don't think it shouldn't include Play - I think that the guide can exist in the way it's been rewritten, but as a "Human task orchestration" guide, there is value in _also_ explaining the way information moves between Operate and Tasklist, not only performing validation. There are no other getting started guides that explore these skills, and it matches the structure of the "Human Workflow" Academy guide.

volodymyr-melnykc commented:
@mesellings @conceptualshark @HanselIdes 
I agree that it depends on the goal we want to achieve with this guide. 

If the goal is to help users validate a designed process quickly, then showcasing it with Play would be the best.
If the goal is to educate users on building human-centric processes, then highlighting Tasklist and Operate is a must. The user should know that Tasklist is the interface where the actual task review and completion happens, and Operate is the tool where the user can monitor the progress of the process.

In my opinion, this guide should be a foundation for educating users on building human-centric processes. This doesn't mean that it shouldn't mention Play, though quick process validation is not a primary objective here.

HanselIdes commented:
Hi @melanie-butcher, I'm bringing you into this conversation as the head of the rapid adoption product stream. I believe you own the getting started guides.

We need a decision about whether the purpose of the HTO Getting Started guide. Is it to:
* show the development process in Web Modeler + Play, and mention Tasklist and Operate
* show the production environment in Web Modeler + Tasklist + Operate, and only mention Play
This is an important branch point, and we don't have alignment.

akeller commented:
What is the status of this PR?

mesellings commented:
@akeller Blocked currently as we await a decision on which way the HTO lead(s) want to go - Slack thread here: https://camunda.slack.com/archives/C0259E9ETNJ/p1730708865839529. I will give a nudge in the thread.

christinaausley commented:
@mesellings is this now slated for 8.7?

mesellings commented:
> @mesellings is this now slated for 8.7?

Yes thanks @christinaausley , this was unblocked last week, so will bring it back into the pipeline for completion by/for 8.7

akeller commented:
I don't see much progress on this PR, so I'm assuming it's not going out with alpha4. If this is incorrect, please be prepared to adjust it for the upcoming [working mode change](https://camunda.slack.com/archives/C026U8GBNSW/p1738938276012929).

akeller commented:
I'm visiting all 8.7 tagged items to make you aware of the [working mode change](https://camunda.slack.com/archives/C026U8GBNSW/p1738938735125789?thread_ts=1738938276.012929&cid=C026U8GBNSW).

> Version 8.7: will be sourced from the `/versioned_docs/version-8.7` folder; URL paths will look like `/docs/8.7/path/to/doc/`.

Please ensure your changes are in the right directory with respect to the version you want to update. Changes currently in this PR would only happen in 8.8. You may want to backport to 8.7.

akeller commented:
Can I get a status update on this PR? How realistic is it that it will move forward and be ready for 8.7?

HanselIdes commented on docs/guides/getting-started-orchestrate-human-tasks.md:
```diff
@@ -118,116 +175,112 @@ Variables are part of a process instance and represent the data of the instance.
-Don't worry about saving your process diagram. Modeler automatically saves every change you make.
+
+- You do not need to save your process diagram as Web Modeler automatically saves every change you make.
+- You can also create [milestones](/components/modeler/web-modeler/milestones.md) to save a version snapshot of your BPMN or DMN diagram at any time.
```
NOTE: Milestones have been renamed to versions within the last few months

HanselIdes commented:
@mesellings I added only one small comment about milestones being renamed to versions, otherwise you have my approval since October

mesellings commented:
@akeller Removed 8.7 as a target, we're not going to get around to including this in 8.7

christinaausley commented:
@mesellings Is this still planned, for 8.8 for example, or still deprioritized?
